<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>832</width>
    <height>700</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>832</width>
    <height>700</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>832</width>
    <height>700</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Maze Generator</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QGraphicsView" name="graphicsView">
      <property name="minimumSize">
       <size>
        <width>500</width>
        <height>500</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>500</width>
        <height>500</height>
       </size>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_maze">
       <attribute name="title">
        <string>Maze</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_7">
        <item row="1" column="0">
         <widget class="QFrame" name="frame_4">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>Width:</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_14" stretch="1,0">
             <property name="topMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QSlider" name="slider_maze_width">
               <property name="cursor">
                <cursorShape>OpenHandCursor</cursorShape>
               </property>
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>50</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="line_maze_width">
               <property name="text">
                <string>10</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Height:</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_13" stretch="1,0">
             <property name="topMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QSlider" name="slider_maze_height">
               <property name="cursor">
                <cursorShape>OpenHandCursor</cursorShape>
               </property>
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>50</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="line_maze_height">
               <property name="text">
                <string>10</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_15">
             <property name="topMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QPushButton" name="but_maze_generate">
               <property name="minimumSize">
                <size>
                 <width>90</width>
                 <height>30</height>
                </size>
               </property>
               <property name="cursor">
                <cursorShape>PointingHandCursor</cursorShape>
               </property>
               <property name="styleSheet">
                <string notr="true">QPushButton {
border-radius: 10px;
border: 2px solid rgb(126, 44, 61); 
background-color: black;
color: gold;
font-family: &quot;Comic Sans MS&quot;;
font: 15px;
}

QPushButton:hover {
	border: 2px solid black;
	font: bold 13px;
	
}</string>
               </property>
               <property name="text">
                <string>Generate</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_4">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QFrame" name="frame_5">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_16">
           <item>
            <widget class="QPushButton" name="but_maze_load">
             <property name="minimumSize">
              <size>
               <width>70</width>
               <height>30</height>
              </size>
             </property>
             <property name="cursor">
              <cursorShape>PointingHandCursor</cursorShape>
             </property>
             <property name="styleSheet">
              <string notr="true">QPushButton {
border-radius: 10px;
border: 2px solid rgb(126, 44, 61); 
background-color: black;
color: gold;
font-family: &quot;Comic Sans MS&quot;;
font: 15px;
}

QPushButton:hover {
	border: 2px solid black;
	font: bold 13px;
	
}</string>
             </property>
             <property name="text">
              <string>Load</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="but_maze_save">
             <property name="minimumSize">
              <size>
               <width>70</width>
               <height>30</height>
              </size>
             </property>
             <property name="cursor">
              <cursorShape>PointingHandCursor</cursorShape>
             </property>
             <property name="styleSheet">
              <string notr="true">QPushButton {
border-radius: 10px;
border: 2px solid rgb(126, 44, 61); 
background-color: black;
color: gold;
font-family: &quot;Comic Sans MS&quot;;
font: 15px;
}

QPushButton:hover {
	border: 2px solid black;
	font: bold 13px;
	
}</string>
             </property>
             <property name="text">
              <string>Save</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_5">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item row="2" column="0">
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_cave">
       <attribute name="title">
        <string>Cave</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="3" column="0">
         <widget class="QFrame" name="frame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Iteration:</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_11">
             <property name="topMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QPushButton" name="but_cave_auto">
               <property name="minimumSize">
                <size>
                 <width>70</width>
                 <height>30</height>
                </size>
               </property>
               <property name="cursor">
                <cursorShape>PointingHandCursor</cursorShape>
               </property>
               <property name="styleSheet">
                <string notr="true">QPushButton {
border-radius: 10px;
border: 2px solid rgb(126, 44, 61); 
background-color: black;
color: gold;
font-family: &quot;Comic Sans MS&quot;;
font: 15px;
}

QPushButton:hover {
	border: 2px solid black;
	font: bold 13px;
	
}</string>
               </property>
               <property name="text">
                <string>Auto</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="but_cave_step">
               <property name="minimumSize">
                <size>
                 <width>70</width>
                 <height>30</height>
                </size>
               </property>
               <property name="cursor">
                <cursorShape>PointingHandCursor</cursorShape>
               </property>
               <property name="styleSheet">
                <string notr="true">QPushButton {
border-radius: 10px;
border: 2px solid rgb(126, 44, 61); 
background-color: black;
color: gold;
font-family: &quot;Comic Sans MS&quot;;
font: 15px;
}

QPushButton:hover {
	border: 2px solid black;
	font: bold 13px;
	
}</string>
               </property>
               <property name="text">
                <string>Step</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_3">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>delay, ms:</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_10" stretch="1,0">
             <property name="topMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QSlider" name="slider_cave_delay">
               <property name="cursor">
                <cursorShape>OpenHandCursor</cursorShape>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="line_cave_delay">
               <property name="text">
                <string>100</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QFrame" name="frame_3">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_12">
           <item>
            <widget class="QPushButton" name="but_cave_load">
             <property name="minimumSize">
              <size>
               <width>70</width>
               <height>30</height>
              </size>
             </property>
             <property name="cursor">
              <cursorShape>PointingHandCursor</cursorShape>
             </property>
             <property name="styleSheet">
              <string notr="true">QPushButton {
border-radius: 10px;
border: 2px solid rgb(126, 44, 61); 
background-color: black;
color: gold;
font-family: &quot;Comic Sans MS&quot;;
font: 15px;
}

QPushButton:hover {
	border: 2px solid black;
	font: bold 13px;
	
}</string>
             </property>
             <property name="text">
              <string>Load</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="but_cave_save">
             <property name="minimumSize">
              <size>
               <width>70</width>
               <height>30</height>
              </size>
             </property>
             <property name="cursor">
              <cursorShape>PointingHandCursor</cursorShape>
             </property>
             <property name="styleSheet">
              <string notr="true">QPushButton {
border-radius: 10px;
border: 2px solid rgb(126, 44, 61); 
background-color: black;
color: gold;
font-family: &quot;Comic Sans MS&quot;;
font: 15px;
}

QPushButton:hover {
	border: 2px solid black;
	font: bold 13px;
	
}</string>
             </property>
             <property name="text">
              <string>Save</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QFrame" name="frame_2">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <item>
              <widget class="QLabel" name="label_8">
               <property name="text">
                <string>width:</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="1,0">
               <item>
                <widget class="QSlider" name="slider_cave_width">
                 <property name="cursor">
                  <cursorShape>OpenHandCursor</cursorShape>
                 </property>
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>50</number>
                 </property>
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="line_cave_width">
                 <property name="text">
                  <string>10</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QLabel" name="label_9">
               <property name="text">
                <string>height:</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="1,0">
               <item>
                <widget class="QSlider" name="slider_cave_height">
                 <property name="cursor">
                  <cursorShape>OpenHandCursor</cursorShape>
                 </property>
                 <property name="minimum">
                  <number>1</number>
                 </property>
                 <property name="maximum">
                  <number>50</number>
                 </property>
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="line_cave_height">
                 <property name="text">
                  <string>10</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QLabel" name="label">
               <property name="text">
                <string>birth limit:</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="1,0">
               <item>
                <widget class="QSlider" name="slider_cave_birth">
                 <property name="cursor">
                  <cursorShape>OpenHandCursor</cursorShape>
                 </property>
                 <property name="maximum">
                  <number>7</number>
                 </property>
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="line_cave_birth">
                 <property name="text">
                  <string>4</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>death limit:</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_8" stretch="1,0">
               <property name="topMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QSlider" name="slider_cave_death">
                 <property name="cursor">
                  <cursorShape>OpenHandCursor</cursorShape>
                 </property>
                 <property name="maximum">
                  <number>7</number>
                 </property>
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="line_cave_death">
                 <property name="text">
                  <string>4</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>initial probability:</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_9" stretch="1,0">
               <property name="topMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QSlider" name="slider_cave_init">
                 <property name="cursor">
                  <cursorShape>OpenHandCursor</cursorShape>
                 </property>
                 <property name="maximum">
                  <number>100</number>
                 </property>
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="line_cave_init">
                 <property name="text">
                  <string>50</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_6">
               <property name="topMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QPushButton" name="but_cave_generate">
                 <property name="minimumSize">
                  <size>
                   <width>90</width>
                   <height>30</height>
                  </size>
                 </property>
                 <property name="cursor">
                  <cursorShape>PointingHandCursor</cursorShape>
                 </property>
                 <property name="styleSheet">
                  <string notr="true">QPushButton {
border-radius: 10px;
border: 2px solid rgb(126, 44, 61); 
background-color: black;
color: gold;
font-family: &quot;Comic Sans MS&quot;;
font: 15px;
}

QPushButton:hover {
	border: 2px solid black;
	font: bold 13px;
	
}</string>
                 </property>
                 <property name="text">
                  <string>Generate</string>
                 </property>
                </widget>
               </item>
               <item>
                <spacer name="horizontalSpacer">
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>40</width>
                   <height>20</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item row="4" column="0">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
